src/story_protocol_python_sdk/resources/Permission.py

```diff
@@ -24,75 +27,182 @@ def __init__(self, web3: Web3, account, chain_id: int):
+        :param signer str: The address that can call `to` on behalf of the `ip_id`.
+        :param to str: The address that can be called by the `signer`.
+        :param permission int: The new permission level.
+        :param func str: [Optional] The function selector string of `to` that can be called by the `signer` on behalf of the `ipAccount`.
+        :param tx_options dict: [Optional] The transaction options.
+        :return dict: A dictionary with the transaction hash and success status if waiting for transaction.
+        """
+        try:
+            validate_address(signer)
+            validate_address(to)
             
-    #         if not self._is_registered(ip_asset):
-    #             raise ValueError(f"The IP account with id {ip_asset} is not registered.")
-
-    #         config_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'scripts', 'config.json'))
-    #         with open(config_path, 'r') as config_file:
-    #             config = json.load(config_file)
-    #         contract_address = None
-    #         for contract in config['contracts']:
-    #             if contract['contract_name'] == 'AccessController':
-    #                 contract_address = contract['contract_address']
-    #                 break
-    #         if not contract_address:
-    #             raise ValueError(f"Contract address for AccessController not found in config.json")
-    #         abi_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'abi', 'AccessController', 'AccessController.json'))
-    #         with open(abi_path, 'r') as abi_file:
-    #             abi = json.load(abi_file)
-
-    #         contract = self.web3.eth.contract(address=contract_address, abi=abi)
-            
-    #         data = contract.encode_abi(
-    #             fn_name="setPermission",
-    #             args=[
-    #                 ip_asset,
-    #                 signer,
-    #                 to,
-    #                 func,
-    #                 permission
-    #             ]
-    #         )
-
-    #         response = self.ip_account.execute(
-    #             to=contract_address,
-    #             value=0,
-    #             account_address=ip_asset,
-    #             data=data
-    #         )
-
-    #         return {
-    #             'txHash': response['txHash']
-    #         }
-
-    #     except Exception as e:
-    #         raise e
-
-    # def _is_registered(self, ip_id: str) -> bool:
-    #     """
-    #     Check if an IP is registered.
-
-    #     :param ip_id str: The IP ID to check.
-    #     :return bool: True if registered, False otherwise.
-    #     """        
-    #     return self.ip_asset_registry_client.isRegistered(ip_id)
\ No newline at end of file
+            self._check_is_registered(ip_id)
+            
+            data = self.access_controller_client.contract.encode_abi(
```

Comment:
I noticed that the `self.access_controller_client.contract.encode_abi` code is duplicated, so perhaps we can try extracting some common functionality, I suggest something like this:

```
def _prepare_and_execute_transaction(self, contract_address, abi_element, args, ip_id, value=0, tx_options=None):
    data = self.access_controller_client.contract.encode_abi(
        abi_element_identifier=abi_element,
        args=args
    )
    
    response = self.ip_account.execute(
        to=contract_address,
        value=value,
        ip_id=ip_id,
        data=data,
        tx_options=tx_options
    )
    
    return {'tx_hash': response['tx_hash']}
    ```